src/components/AlbumsList.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import Snackbar from '@mui/material/Snackbar';
+import {
+  Container,
+  TextField,
+  Button,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  Typography,
+  Box,
+  Grid,
+} from '@mui/material';
+
+const AlbumsList = () => {
+  const [albums, setAlbums] = useState([]); // for albums
+  const [limit, setLimit] = useState(1); // for pagination
+  const [searchType, setSearchType] = useState('albumName'); //default value for select input
+  const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
+  const [message, setMessage] = useState(''); // for not found message
+  const [errorMessage, setErrorMessage] = useState(''); // for error message
+
+  //make an API call with search values to backend and return the result
+  const fetchAlbums = async (searchType, searchTerm, limit) => {
+    try {
+      const response = await axios.get(
+        'http://localhost:8000/api/v1/albums/filter',
+        {
+          params: {
+            limit,
+            [searchType]: searchTerm,
+          },
+        }
+      );
+      const searchResult = response.data.albums;
+      setAlbums(searchResult);
+      if (searchResult.length === 0) {
+        setMessage('No result found');
+      }
+    } catch (error) {
+      console.error('Error fetching albums:', error);
+      setErrorMessage(error.message);
+      setOpen(true);
+    }
+  };
+
```

Comment:
Oh I also meant to add earlier; it would be nice to just have some initial values shown, when the user comes to this screen without having them have to press search first. We could do that by calling `fetchAlbums` as soon as this component is loaded, with no parameters. We can use `useEffect` to do something when the component mounts, by passing in an empty array as the second argument: https://medium.com/devil-is-in-the-details/dependency-array-in-useeffect-hook-d73e0ef2ab33

```js
useEffect(() => {
  fetchAlbums('albumName', '', 10)
}, []);

Reply:
Yes, that is something I wanted to add too, but I was waiting for AlbumGrid to merge first. I will add that